liveblocks/liveblocks
# feat(publish): publish from CI

## What does this PR do?

This PR enables publishing of packages from the CI using `workflow_dispatch`

**Check out a successful run [here](https://github.com/liveblocks/liveblocks/actions/runs/4266706391/jobs/7427540895)!**

## Where should the reviewer start?

The process happens in 2 places.

### 1. Running a release
When merged on `main`, it will be done through the github UI with a `tag` input:
![actions-manually-run-workflow](https://user-images.githubusercontent.com/6947265/221310036-24376287-d978-400a-8913-cea3e0003067.jpg)

Though for now it must be done through the `gh` CLI:
`gh workflow run "Publish packages" --ref feat/automatic-deploy -f tag=<GITHUB_TAG>`

### 2. Workflow takes over
1. Install all dependencies
2. Setup the release `release.sh`
3. Build packages
4. Publish the release (`publish.sh`) and pushes to the branch
<img width="952" alt="Screen Shot 2023-02-24 at 5 39 41 PM" src="https://user-images.githubusercontent.com/6947265/221310551-f46cb4fd-c589-4db9-9dbb-e79912026c87.png">
5. Creates & pushes the release tag and prompts for release notes
<img width="831" alt="Screen Shot 2023-02-24 at 5 35 16 PM" src="https://user-images.githubusercontent.com/6947265/221308207-c4284b3b-35bf-4691-b789-f07ad5532807.png">

## How should this be manually tested?

Run `gh workflow run "Publish packages" --ref feat/automatic-deploy -f tag=<GITHUB_TAG>`
Check that action runs
Check that packages are properly published
Check that `beta` npm tag won't work, ex: `gh workflow run "Publish packages" --ref feat/automatic-deploy -f tag=1.0.2-beta0` since its a topic branch
Check that release without a tag won't work, ex: `gh workflow run "Publish packages" --ref feat/automatic-deploy -f tag=1.0.2` since its not `main` branch

## What are the relevant tickets?

https://github.com/liveblocks/liveblocks.io/issues/891

## Further discussion
Next step will be to use it for schema packages, it will require setting up some modularity, ex: repo url, package paths
